controllers/metal3.io/preprovisioningimage_controller.go

```diff
@@ -110,6 +110,13 @@ func (r *PreprovisioningImageReconciler) Reconcile(ctx context.Context, req ctrl
 		log.Info("requeuing to check for secret", "after", delay)
 		result.RequeueAfter = delay
 	}
+
+	notReady := imageprovider.ImageNotReady{}
+	if errors.As(err, &notReady) {
+		log.Info("image is not ready yet, requeuing", "after", notReady.RequeueAfter)
+		result.RequeueAfter = notReady.RequeueAfter
```

Comment:
A potential counter-argument is that it's the implementation, not the controller, that knows how long to wait.

Reply:
This is similar to the BMH controller, where the provisioner tells the controller what the requeue delay should be.

You probably want to set a minimum value here. Otherwise, you must always set `result.Requeue = true` otherwise a `RequeueAfter` of 0 (the default!) will not result in a requeue and the CR will ~never be updated again.

Reply:
Indeed, result and error are kept as a separate return arguments